app/models/cafe_work.rb

```diff
@@ -0,0 +1,248 @@
+# encoding: UTF-8
+class CafeWork < ActiveRecord::Base
+  # Associations
+  belongs_to :profile
+  has_and_belongs_to_many :councils
+
+  # Validations
+  validates :work_day, :pass, :lp, :lv, presence: true
+  validates :pass, :lp, inclusion: { in: 1..4 }
+  validates :lv, inclusion: { in: 1..7 }
+  validates :name, :lastname, :phone, :email, presence: true, if: :has_worker?
+  validates :pass, uniqueness: { scope: [:work_day, :lv, :lp, :d_year] }
+
+  # Scopes
+  scope :with_worker, -> { where('profile_id IS NOT null OR access_code IS NOT null') }
+  scope :between, ->(from, to) { where(work_day: from..to) }
+  scope :ascending, -> { order(pass: :asc) }
+  scope :week, ->(week) { where(lv: week) }
+  scope :period, ->(p) { where(lp: p) }
+  scope :year, ->(y) { where(d_year: y) }
+
+  after_update :send_email, if: :has_worker?
+
+  # A custom class for the worker
+  # For more information see Assignee model or
+  # read comments above methods
+  def worker
+    @worker || Assignee.new(worker_attributes)
+  end
+
+  # Sends email to worker
+  # /d.wessman
+  def send_email
+    CafeMailer.sign_up_email(self).deliver_now
+  rescue
+    Rails.logger.info 'Mailer could not connect, rescued here'
+  end
+
+  # Prepares work for a user to sign up, without saving
+  # Will only change attributes if there is no worker and
+  # user is present.
+  # /d.wessman
+  def load(user)
+    self.attributes = worker.load_profile(user)
+  end
+
+  # Shows different status texts depending on the user.
+  # /d.wessman
+  def status_text(user)
+    case status_view(user)
```

Comment:
How should I do then? Should I return a symbol or something then? 
Like 
- :sign_up
- :edit
- :authorize
- :assigned
? 😄 @henrikssn